connectors.go

```diff
@@ -77,6 +77,14 @@ type ObjectMetadataConnector interface {
 	ListObjectMetadata(ctx context.Context, objectNames []string) (*ListObjectMetadataResult, error)
 }
 
+// MetadataUpsertConnector is an interface that extends the Connector interface with
+// the ability to create metadata.
+type MetadataUpsertConnector interface {
```

Comment:
can this be UpsertMetadataConnector? All the other interfaces are VerbNounConnector